geps/gep-1709.md

```diff
@@ -266,6 +285,25 @@ VERSION file contains the semver of the latest release and is manually bumped at
 release time. The script hack/verify-all.sh will be updated to ensure the
 generated `.go` file is up to date with the VERSION file.
 
+### Multiple reports merge
```

Comment:
Merging seems like it could get pretty complex. If different `modes` have different supported features are we just going to take the intersection of them? What about failing tests?

For GKE, I think I'd rather have different files for each unique mode. In our case it would be per GatewayClass, and we already have [8 unique GatewayClasses that we support](https://cloud.google.com/kubernetes-engine/docs/how-to/gatewayclass-capabilities).

I think I'd rather take one step closer to [upstream k8s conformance reporting structure](https://github.com/cncf/k8s-conformance/tree/master/v1.28) here and end up with something like this:

```
- v1.0
  - gke
    - README.md
    - mode-1-report.yaml
    - mode-2-report.yaml
```

We could potentially take this one step further and pull some of the product specific information out of the report and into a separate `product.yaml` (eg: https://github.com/cncf/k8s-conformance/blob/master/v1.28/gke/PRODUCT.yaml).

I don't know, I'm probably missing some of the goals here, so let me know if these recommendations don't make sense.

Reply:
> Merging seems like it could get pretty complex. If different modes have different supported features are we just going to take the intersection of them? What about failing tests?

The idea is to have a section for each mode running a specific implementation version. If different modes have different supported features, we create two different sections, one per mode, along with the `supportedFeatures` and the results (successful, failed, skipped tests).

>For GKE, I think I'd rather have different files for each unique mode. In our case it would be per GatewayClass, and we already have [8 unique GatewayClasses that we support](https://cloud.google.com/kubernetes-engine/docs/how-to/gatewayclass-capabilities).
>
>I think I'd rather take one step closer to [upstream k8s conformance reporting structure](https://github.com/cncf/k8s-conformance/tree/master/v1.28) here and end up with something like this:
>
>```- v1.0
>  - gke
>    - README.md
>    - mode-1-report.yaml
>    - mode-2-report.yaml
>```
> We could potentially take this one step further and pull some of the product specific information out of the report and into a separate product.yaml (eg: https://github.com/cncf/k8s-conformance/blob/master/v1.28/gke/PRODUCT.yaml).

I get your point, and I agree that with the proposed solution we can end up having a gigantic `yaml` file. I like the approach of having a "summary report file" and a set of reports for each mode/version pair (or something similar).
I'll put something together that takes into account these considerations.

Reply:
@robscott it took some time, but eventually, I came up with a proposal very similar to what you suggested. Let me know what you think 😄

Reply:
Thanks @mlavacca! Sorry for the delay on reviewing this!